Lib/test/test_ssl.py

```diff
@@ -3531,6 +3531,10 @@ def serve():
         server.close()
         # Sanity checks.
         self.assertIsInstance(remote, ssl.SSLSocket)
+
+        with self.assertRaises(NotImplementedError):
+            remote.dup()
```

Comment:
Maybe this is not the best place to put this check, but it's not obvious to construct the `ssl.SSLSocket` instance manually, and there is no other place in this test file where we are sure to have such instance.